screens/Budget/UpdateBudgetScreen.tsx

```diff
@@ -0,0 +1,199 @@
+import Button from "../../components/Button";
+import { ControlledInputDecimal, InputDecimal } from "../../components/InputDecimal";
+
+
+
+export default function UpdateBudgetScreen({ navigation, route }: RootStackScreenProps<'UpdateBudget'>) {
+    const [passwordHash, setpasswordHash] = useState('');
+    const [amount, setAmount] = useState(0);
+    const [amountLable, setAmountLabel] = useState('')
+    const [BudgetCategory, setBudgetCategory] = useState<BudgetCategory | null>(null);
+
+    const [updateBudget] = useMutation<UpdateBudgetCategoryMutation>(UpdateBudgetCategoryDocument, {
+        variables: { passwordHash, id: BudgetCategory?.id, amount },
+        onError: (error => {
+            alert(error.message);
+        }),
+        onCompleted: ((response) => {
+            if (response.updateBudgetCategory.__typename == 'BudgetCategorySuccess') {
+                // navigation.goBack();
+                navigation.navigate("Root", { screen: "Budget", params: { refresh: (Math.random()) } });
+            }
+        })
+    })
+
+    const [deleteBudget] = useMutation<DeleteBudgetCategoryMutation>(DeleteBudgetCategoryDocument, {
+        variables: { passwordHash, id: BudgetCategory?.id },
+        onCompleted: ((response) => {
+            if (response.deleteBudgetCategory.__typename == 'DeleteSuccess') {
+                // navigation.goBack();
+                // navigation.navigate("Root", { screen: "Budget", params: { refresh: true } });
+                navigation.navigate("Root", { screen: "Budget", params: { refresh: (Math.random()) } });
+            }
+        })
+    })
+
+    useEffect(() => {
+        getData();
+    }, []);
+
+    const getData = async () => {
+        try {
+            const value = await AsyncStorage.getItem('passwordHash')
+            if (value != null) {
+                setpasswordHash(value);
+            }
+        } catch (e) {
+            setpasswordHash('undefined');
+        }
+        setBudgetCategory(route.params.budgetCategory)
+        setAmount(route.params.budgetCategory.amount)
+        setAmountLabel(route.params.budgetCategory.amount.toFixed(2))
+    }
+
+    return (
+        <View style={styles.screen}>
+            <View style={styles.amountInputContainer}>
+                <View style={styles.dollarSignAndAmountInput}>
+                    <Text style={styles.dollarSign}>$</Text>
+                    <ControlledInputDecimal
+                        callbackNumber={setAmount}
+                        label={amountLable}
+                        callbackLable={setAmountLabel}
+                        placeholder='Enter Amount'
+                    />
+                </View>
+                <View>
+
+                </View>
+            </View>
+            <View style={styles.detailsRow}>
+                <View style={styles.detailsIconAndLabel}>
+                    <Text style={styles.fieldLabel}>For Budget:</Text>
```

Comment:
Implement this (ie. updating a budget category should display the correct 'For Budget').